src/components/Records/Pagination.vue

```diff
@@ -43,80 +43,88 @@
-    */
-    export default {
-        name: "Pagination",
-        props: {
-            totalPages: {
-                type: Number,
-                default: 0
-            },
-        },
-        data() {
-            return {
-                currentPage: null
-            }
-        },
-        watch: {
-            '$route.name': function () {
-                this.currentPage = 1;
-            },
-            '$route.query': function (newVal) {
-                let _module = this;
-                if (!Object.prototype.hasOwnProperty.call(newVal, "page")) {
-                    _module.currentPage = 1;
-                } else {
-                    _module.currentPage = Number(newVal.page);
-                }
-            }
-        },
-        created() {
-            let currentPage = this.$route.query.page;
-            currentPage !== undefined ? this.currentPage = Number(currentPage) : this.currentPage = 1;
-        },
-        methods: {
-          /**
-           * Set the current query page parameter value to the given input for vueJs router
-           * @param {Number} pageNumber - the page to go to
-           */
-            paginate: async function (pageNumber) {
-                if (pageNumber !== this.currentPage) {
-                    let _module = this;
-                    let currentQuery = {};
-                    _module.currentPage = pageNumber;
-                    Object.keys(_module.$route.query).forEach(function (param) {
-                      currentQuery[param] = _module.$route.query[param]
-                    });
-                  currentQuery.page = pageNumber;
-                    await _module.$router.push({
-                        name: _module.$route.name,
-                        query: currentQuery
-                    });
-                }
-            },
-          /**
-           * Set the current page to the first page
-           */
-            first: function () {
-                this.paginate(1);
-            },
-          /**
-           * Set the current page to the last page
-           */
-            last: function () {
-                this.paginate(this.totalPages);
-            }
-        },
-    }
+  /** Component to handle the advanced search filters for the searchFairsharingRecords query.
+   * @vue-prop {Number} [totalPages = 0] - the total number of pages to display
+   * @vue-data {Number} [currentQuery = null] - the current page number
+   */
+  export default {
+    name: "Pagination",
+    props: {
+      totalPages: {
+        type: Number,
+        default: 0
+      },
+    },
+    data() {
+      return {
+        currentPage: null,
+        allowPaginate: true
+      }
+    },
+    watch: {
+      '$route.name': function () {
+        this.currentPage = 1;
+      },
+      '$route.query': function (newVal) {
+        let _module = this;
+        if (!Object.prototype.hasOwnProperty.call(newVal, "page")) {
```

Comment:
I recently figured out an easier wait of checking for properties:
Object.keys(myVar).includes("myField").
Im not asking for a change. This is just an informative note.

Reply:
that's better and newer. I changed it to include.

Reply:
I did change it and it caused some problem so ATM I stuck with the same old method.